src/proto/serializer.ts

```diff
@@ -36,15 +36,30 @@ export class ProtoSerializer {
             Signature: Buffer.from(transaction.getSignature().hex(), "hex")
         });
 
-        if ( transaction.getOptions().valueOf() !== TRANSACTION_OPTIONS_DEFAULT ) {
+        if (transaction.getOptions().valueOf() !== TRANSACTION_OPTIONS_DEFAULT) {
             protoTransaction.Options = transaction.getOptions().valueOf();
         }
-        
-        let encoded = proto.Transaction.encode(protoTransaction).finish();
-        let buffer = Buffer.from(encoded);
+
+        if (transaction.getOptions().isGuardedTx()) {
+            const guardianAddress = this.tryGetGuardianAddress(transaction.getGuardian());
+            protoTransaction.GuardAddr = new Address(guardianAddress.bech32()).pubkey();
+            protoTransaction.GuardSignature = Buffer.from(transaction.getGuardianSignature().hex(), "hex");
+        }
+
+        const encoded = proto.Transaction.encode(protoTransaction).finish();
+        const buffer = Buffer.from(encoded);
+
         return buffer;
     }
 
+    private tryGetGuardianAddress(guardianAddress: IAddress | undefined): IAddress {
```

Comment:
Given the comment from `transaction.ts`, I don't think `getGuardian` can return `undefined`. This method could be unnecessary, see comment below

Reply:
true,
removed it
:+1: